dbx/utils/common.py

```diff
@@ -69,17 +76,56 @@ def get_context() -> Any:
             return None
 
 
-class DeploymentFile:
+class AbstractDeploymentConfig(ABC):
```

Comment:
This ABC `AbstractDeploymentConfig` and the concrete classes `JsonDeploymentConfig` and `YamlDeploymentConfig` is the core of the logic changes.